src/fortune-wheel-booth-frontend/src/App.tsx

```diff
@@ -1,31 +1,163 @@
+    await new Promise((resolve) => {
+      authClient.login({
+        identityProvider:
+          process.env.DFX_NETWORK === 'ic'
+            ? 'https://identity.ic0.app'
+            : `http://rdmx6-jaaaa-aaaaa-aaadq-cai.localhost:4943/`,
+        onSuccess: resolve,
+      });
+    });
+
+    const identity = authClient.getIdentity();
+    setupIcState(identity);
+  }, [setupIcState]);
+
+  useEffect(() => {
+    (async () => {
+      const authClient = await AuthClient.create();
+
+      const authenticated = await authClient.isAuthenticated();
+      if (authenticated) {
+        setupIcState(authClient.getIdentity());
+      } else {
+        resetIcState();
+      }
+    })();
+  }, [setupIcState, resetIcState]);
+
+  const extractPrize = async (text: string) => {
+    const userPrincipal: Principal = Principal.fromText(text);
+    if (adminActor) {
+      try {
+        const extraction = await adminActor.extract(userPrincipal);
+        console.log(extraction);
+      } catch (error: any) {
+        console.log(error);
+        if (error.message.includes('Only admins can extract')) {
+          logout();
+          setCanisterErrorResponse('Only admins can extract');
+        }
+      }
+    }
+  };
+
+  const logout = useCallback(async () => {
+    if (adminActor) {
+      await (await AuthClient.create()).logout();
+      resetIcState();
+    }
+  }, [adminActor, resetIcState]);
+
+  if (!adminPrincipal) {
+    return <div>Loading...</div>;
   }
 
+  const isAnonymous = adminPrincipal.isAnonymous();
+
   return (
-    <main>
-      <img src="/logo2.svg" alt="DFINITY logo" />
-      <br />
-      <br />
-      <form action="#" onSubmit={handleSubmit}>
-        <label htmlFor="name">Enter your name: &nbsp;</label>
-        <input id="name" alt="Name" type="text" />
-        <button type="submit">Click Me!</button>
-      </form>
-      <section id="greeting">{greeting}</section>
-    </main>
+    <>
+      {isAnonymous && (
+        <div className='flex justify-center items-center h-full w-full flex-col gap-4'>
+          <img
+            className='absolute top-10 left-0 right-0 m-auto h-32 z-20'
+            src={icpLogo}
+            alt='icp logo'
+          />
+          {canisterErrorResponse && (
+            <p className='text-red-500 text-sm absolute top-1/4 left-0 right-0 m-auto text-center'>
+              {canisterErrorResponse}
+            </p>
+          )}
+          <button
+            className='bg-white rounded-xl shadow-sm w-44 text-center px-4 py-2'
+            onClick={handleLogin}
+          >
+            <p className='text-xl'>Admin</p>
+          </button>
+          <a
+            className='bg-white rounded-xl shadow-sm w-44 text-center px-4 py-2 text-xl'
+            href='/fortune-wheel/'
+          >
+            Fortune Wheel
+          </a>
+        </div>
+      )}
+      {!isAnonymous && (
+        <>
+          <Scanner
+            onResult={(text) => extractPrize(text)}
```

Comment:
This callback is called many times even for just one scan, we need to limit it to just one